docs/DeveloperGuide.md

```diff
@@ -1,34 +1,513 @@
+
+The following activity diagram summarizes what happens when a user executes a new command:
+
+![cap uml diagram](Images/CalculateCapSequence.png)
+
+### 4.5 Reminder Feature
+
+The proposed reminder mechanism is facilitated by `ReminderAction`. It extends `Action` and the output is passed onto `UI` for display. Additionally, it implements the following operations:
+
+* `ReminderAction#act()`- List out the deadlines and events tasks that are due within 3 days
+
+Given below is an example usage scenario and how the reminder mechanism behaves at each step.
+
+Step 1. The user executes `reminder` command to list out tasks due within 3 days. Command is then parsed by `ReminderAction#act()`.
+
+Step 2. `ReminderAction#act` retrieves tasklist data from the user's list
+
+Step 3. `ReminderAction#act` then sorts the due dates in ascending order
+
+Step 4: Tasks due within 3 days are returned to the user through Ui
+
+The following sequence diagram diagram shows how the reminder operation works
+
+![Reminder_Sequence_Diagram](Images/ReminderAction_Sequence_Diagram.png)
+
+### 4.6 Remind Feature
+
+Another proposed manual reminder mechanism is facilitated by `RemindAction`. It extends `Action` to execute command given by the user, output are then passed on to `Ui` for display. 
+Additionally, it implements the following operations:
+
+* `RemindAction#act()` - Set the reminder to be executed on the chosen time.
+* `RemindAction#prepare()` - Parse user command to suitable parameter for `RemindAction#act()` function.
+* `RemindAction#getSchedule` - Returns the schedule set by the user.
+
+Given below is an example usage scenario and how the remind mechanism behaves at each step.
+
+Step 1. The user executes `remind [time]` command to set schedule for the reminder. Command is then parsed by `RemindAction#prepare()` to be passed as arguments for `RemindAction#act()`.
+
+Step 2. `RemindAction#act()` calls `RemindAction#getSchedule` to pass the schedule later in `Ui`.
+
+Step 3. The schedule is returned to the user through `Ui`.
+
+The following activity diagram summarizes what happens when a user executes a new command:
+
+![Postpone_Sequence_Diagram](Images/Remind.png)
+
+### 4.7 Snooze Feature
+
+The proposed snooze mechanism is facilitated by `SnoozeAction`. It extends `Action` to execute command given by the user, output are then passed on to `Ui` for display. 
+Additionally, it implements the following operations:
+
+* `RemindAction#getNewInterval` - Returns the new interval set by the user.
+
+Given below is an example usage scenario and how the snooze mechanism behaves at each step.
+
+Step 1. The user executes `snooze` command to snooze for the reminder. 
+
+Step 2. `SnoozeAction#getNewInterval` sets and returns the new interval.
+
+Step 3. The new interval is returned to the user through `Ui`.
+
+The following activity diagram summarizes what happens when a user executes a new command:
+
+![Postpone_Sequence_Diagram](Images/Snooze.png)
```

Comment:
Activation bar of method getNewInterval() starts before the method call arrives in this diagram